R/linelist_merge.R

```diff
@@ -0,0 +1,416 @@
+      output_sheet$lab_resdate<-as.Date(output_sheet$lab_resdate,origin="1899-12-30")
+
+    }
+
+    #Recode this if missing use patcurrent status
+    if(!("patcourse_status" %in% vars)){
+      print("patcourse_status")
+      output_sheet$patcourse_status<-NA
+    }
+
+
+    #ensure this is for death, some files ie. KEN had date of outcome column whihc was used in this place
+    #some cases who were not dead would have this entred into their date of death
+    #make date of death missing if patcourse_status !=dead
+
+    if(!("patcourse_datedeath" %in% vars)){
+      print("patcourse_datedeath")
+      output_sheet$patcourse_datedeath<-NA
+      output_sheet$patcourse_datedeath<-as.Date(output_sheet$patcourse_datedeath,origin="1899-12-30")
+    }
+
+    if(!("patcurrent_status" %in% vars)) {
+      print("patcurrent_status")
+      output_sheet$patcurrent_status<-NA
+}
+
+if(length(grep("patsympt",vars))==0){
+        print("no patsmpt")
+        output_sheet$patsympt_other<-NA
+      }
+
+
+    if(!("patcourse_datedischarge" %in% vars)){
+      print("patcourse_datedischarge")
+      output_sheet$patcourse_datedischarge<-NA
+      output_sheet$patcourse_datedischarge<-as.Date(output_sheet$patcourse_datedischarge,origin="1899-12-30")
+    }
+
+
+    if(!("patinfo_ageonsetunit" %in% vars)){
+      print("patinfo_ageonsetunit")
+      output_sheet$patinfo_ageonsetunit<-NA
+    }
+
+    if(!("patinfo_ageonsetunitdays" %in% vars)){
+      print("patinfo_ageonsetunit")
+      output_sheet$patinfo_ageonsetunitdays<-NA
+    }
+
+
+
+    #keep variables of interest
+    output_sheet<- output_sheet %>% dplyr::select(report_date, patinfo_ageonset, patinfo_ageonsetunit,patinfo_ageonsetunitdays,
+                                                  patinfo_sex, patinfo_resadmin1,
+                                                  patinfo_resadmin2, report_classif,
+                                                  pat_symptomatic, pat_asymptomatic,
+                                                  comcond_preexist1, comcond_preexist,
+                                                  patinfo_occus, expo_travel, expo_travel_country,
+                                                  expo_contact_case, lab_result,
+                                                  lab_datetaken, lab_resdate,
+                                                  patcourse_status,
+                                                  patcourse_datedeath, patcourse_datedischarge, country_iso, contains("patsympt"), patcurrent_status)
+
+
+
+    #add cleaned output sheet to a list
+    output[[f]] <-output_sheet
+  }
+    #merge all cleaned sheets into one
+    output_fin<-output[!sapply(output,is.null)]
+    big_data<-Reduce(function(...) merge(..., all=T), output_fin)
+    big_data<-big_data[order(big_data$country_iso),]
+    # write xlsx file to take on to ext step of cleaning
+    openxlsx::write.xlsx(big_data, here::here("inst",paste0(outputname, Sys.Date(),".xlsx")))
```

Comment:
dont do to inst - this should be to whatever file they specify in the function arguments

Reply:
changed to rio::export